parachain/pallets/control/src/lib.rs

```diff
@@ -473,5 +477,14 @@ pub mod pallet {
 			});
 			Ok(())
 		}
+
+		// Allocate a unique id for a message. Used in the implementation of SendMessage,.
+		pub(crate) fn alloc_message_id() -> H256 {
+			NextMessageId::<T>::mutate(|next_message_id| {
+				*next_message_id = next_message_id.saturating_add(1)
+			});
+			let next_message_id = NextMessageId::<T>::get();
+			sp_io::hashing::blake2_256(&(next_message_id.encode())).into()
+		}
```

Comment:
Prefer we follow the same way xcm used to generate the topicId
https://github.com/Snowfork/polkadot-sdk/blob/0b22697c04cb30d86fc618d3217d771b58ffd734/polkadot/xcm/xcm-builder/src/routing.rs#L48

And moving the generating logic to `outbound_queue` pallet in case upper layer module(e.g. control pallet) does not provide a valid one.

Reply:
Ah, nice find! Will try that approach.

Reply:
Fixed in https://github.com/Snowfork/snowbridge/pull/997/commits/b0e408e27129dff6fcd049dcfb3e7aa836141b46